### ISSUE TITLE
Difficulty with NB16 installer

### ISSUE THREAD

**Issue by pedro-w** [apache/netbeans#5060]
### Apache NetBeans version

Apache NetBeans 16 release candidate

### What happened

I want to install NB16 on Windows 11 from the windows exe installer Apache-NetBeans-16-bin-windows-x64.exe

I currently have Java 17 installed, and on the PATH. Also I have a working installation of NB15.

Running the installer by double clicking failed to install - it could not find any JRE.

![image](https://user-images.githubusercontent.com/22172519/206422015-d080b7f4-35ab-4a98-abc7-db339fd9c494.png)

I re-ran the installed from the command line like this

    .\Apache-NetBeans-16-bin-windows-x64.exe --javahome 'C:\Program Files\Eclipse Adoptium\jdk-17.0.3.7-hotspot\'

Which gave the message: 
Java runtime environment (JRE) not found. There is no JRE at the specified location C:\Program Files\Eclipse Adoptium\jdk-17.0.3.7-hotspot"

![image](https://user-images.githubusercontent.com/22172519/206421889-d3f02a7d-ce03-4cd6-8829-73e6a4fda6c2.png)

The correct invocation was 

    .\Apache-NetBeans-16-bin-windows-x64.exe --javahome 'C:\Program Files\Eclipse Adoptium\jdk-17.0.3.7-hotspot'

which successfully installed NB16.

So I think there may be two bugs
1. Installer needs to improve the JDK detection
2. Path processing shouldn't be sensitive to trailing backslash

(also the fonts on the installer dialog are tiny on my screen but that seems to be par for the course on windows)

### How to reproduce

Attempt to install from the .exe installer as described above.

### Did this work correctly in an earlier version?

Apache NetBeans 15

### Operating System

Windows 11 Pro 22H2 (x64)

### JDK

openjdk version "17.0.3" 2022-04-19 OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7) OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode, sharing)

### Apache NetBeans packaging

Apache NetBeans provided installer

### Are you willing to submit a pull request?

Yes

### Code of Conduct

Yes

**Comment by pedro-w** [comment]
On the trailing backslash things, I think maybe that's not our fault. 

Powershell filename completion (pressing TAB etc. ) automatically puts in single quotes if the path contains a space BUT when it's passed to the program it seems to treat backslash as an escape - almost. If it were an escape I would expect `\'` to be a literal `'` but the program actually gets `"`. I don't really understand this but you can see the double quote appended to the path in my second screenshot. So, yes, there is no JRE at `C:\Program Files\Eclipse Adoptium\jdk-17.0.3.7-hotspot"`

**Comment by ebarboni** [comment]
could it be a registry issue? I always install jdk with all the registry setup maybe the default adoptium doesn't set it.

**Comment by pedro-w** [comment]
Thanks for the response. I've attached the output of `Apache-NetBeans-16-bin-windows-x64.exe --verbose` It does seem to be looking in the registry and also on the path. My understanding was that the registry settings were an Oracle JDK thing and not best practice in general. Also maybe the path required is the system path and not the user path - I definitely have java on my user path.

[installer.txt](https://github.com/apache/netbeans/files/10206417/installer.txt)

**Comment by pedro-w** [comment]
Further to this - I do have Java on the 'System' path and not just on the user path (I'm assuming that, as the installer is running as elevated, it would use the system path?)
Relevant code seems to be 
https://github.com/apache/netbeans/blob/c084119009d2e0f736f225d706bc1827af283501/nbi/engine/native/launcher/windows/src/Launcher.c#L330
and 
https://github.com/apache/netbeans/blob/c084119009d2e0f736f225d706bc1827af283501/nbi/engine/native/launcher/windows/src/JavaUtils.c#L740

**Comment by ebarboni** [comment]
thanks for investigation maybe @lbruun could help. The installer native part will be on his onw repo over there
https://github.com/apache/netbeans-native-installers/tree/master/src/main/cpp/launcher/windows 

I need to setup the build we have a few commit that are since a long time not in the final installer.

**Comment by pedro-w** [comment]
@ebarboni  I was going to take a look at this. I can build nlw.exe from the repo at [apache/netbeans-native-installers](https://github.com/apache/netbeans-native-installers.git) after applying [your PR](https://github.com/apache/netbeans-native-installers/pull/3) but not sure how I could re-bundle a modified version of the exe into the installer to test it. Any hints?

**Comment by ebarboni** [comment]
This will be complicated I guess, but somehow the maven artefact build from [apache/netbeans-native-installers](https://github.com/apache/netbeans-native-installers.git) should be set as external for nbi.engine and build updated to extract the exe in right place and proceed to the build.

currently if I understand the workfow correclty
harness/libs.nbi.engine/build.xml do the copy of binaries to nbi/engine
and 
nbi/engine use the copy not doing any native compile

wouldbe nice if on master branch we could use snapshot from maven apache repo (to refine until we could release the artefact)

**Comment by pedro-w** [comment]
makes sense. If I run the nlw.exe it just says "It`s only the launcher stub." and I guess at some point the main netbeans build will append data onto that file. So we maybe need to put our updated zip from the native installer output into harness/libs.nbi.engine/external replacing what was there.
